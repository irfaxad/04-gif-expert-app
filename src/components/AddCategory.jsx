import { useState } from "react";

export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    onAddCategory(inputValue);
    setinputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar GIFs"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};
