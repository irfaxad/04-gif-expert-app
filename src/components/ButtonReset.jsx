import React from "react";

export const ButtonReset = ({ onReset, label }) => {
  return (
    <button type="button" name="Reset" onClick={onReset}>
      {label}
    </button>
  );
};
