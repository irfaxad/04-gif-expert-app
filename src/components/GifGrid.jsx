import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const [counter, setCounter] = useState(10);

  useEffect(() => {
    getGifs(category);
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <p>{counter}</p>
      <button onClick={() => setCounter(counter + 1)}>+1</button>
    </>
  );
};
