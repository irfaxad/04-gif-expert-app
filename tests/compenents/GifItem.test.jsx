import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("GifItem", () => {
  test("Debe coincidir con el snapshot", () => {
    const { container } = render(<GifItem title="title" url="url" />);
    expect(container).toMatchSnapshot();
  });
});
