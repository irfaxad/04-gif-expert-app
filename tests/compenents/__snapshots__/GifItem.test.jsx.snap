// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="url"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
